docs/source/_notebooks/xgboost-titanic.rst

```diff
@@ -0,0 +1,2608 @@
+            -0.060
+        </td>
+        <td style="padding: 0 0.5em 0 0.5em; text-align: left; border: none;">
+            SibSp
+        </td>
+        
+            <td style="padding: 0 0.5em 0 1em; text-align: right; border: none;">
+                0.000
+            </td>
+        
+    </tr>
+            
+                <tr style="background-color: hsl(0, 100.00%, 94.81%); border: none;">
+        <td style="padding: 0 1em 0 0.5em; text-align: right; border: none;">
+            -0.063
+        </td>
+        <td style="padding: 0 0.5em 0 0.5em; text-align: left; border: none;">
+            &lt;BIAS&gt;
+        </td>
+        
+            <td style="padding: 0 0.5em 0 1em; text-align: right; border: none;">
+                1.000
+            </td>
+        
+    </tr>
+            
+                <tr style="background-color: hsl(0, 100.00%, 92.71%); border: none;">
+        <td style="padding: 0 1em 0 0.5em; text-align: right; border: none;">
+            -0.102
+        </td>
+        <td style="padding: 0 0.5em 0 0.5em; text-align: left; border: none;">
+            Age
+        </td>
+        
+            <td style="padding: 0 0.5em 0 1em; text-align: right; border: none;">
+                19.000
+            </td>
+        
+    </tr>
+            
+                <tr style="background-color: hsl(0, 100.00%, 83.30%); border: none;">
+        <td style="padding: 0 1em 0 0.5em; text-align: right; border: none;">
+            -0.333
+        </td>
+        <td style="padding: 0 0.5em 0 0.5em; text-align: left; border: none;">
+            Pclass=3
+        </td>
+        
+            <td style="padding: 0 0.5em 0 1em; text-align: right; border: none;">
+                1.000
+            </td>
+        
+    </tr>
+            
+    
+            </tbody>
+        </table>
+    
+                
+            
+    
+            
+    
+    
+    
+        
+    
+        
+    
+        
+    
+        
+    
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+    
+    
+
+
+
+
+5. Adding text features
+-----------------------
+
+Right now we treat *Name* field as categorical, like other text
+features. But in this dataset each name is unique, so XGBoost does not
+use this feature at all, because it's such a poor discriminator: it's
+absent from the weights table in section 3.
+
+But *Name* still might contain some useful information. We don't want to
+guess how to best pre-process it and what features to extract, so let's
+use the most general character ngram vectorizer:
+
+.. code:: python
+
+    from sklearn.pipeline import FeatureUnion
+    from sklearn.feature_extraction.text import CountVectorizer
+    
+    vec2 = FeatureUnion([
+        ('Name', CountVectorizer(
+            analyzer='char_wb',
+            ngram_range=(3, 4),
```

Comment:
Could you try it with `(2, 4)` - "Mr" is just two chars, so maybe it could help?